src/test/java/com/cdancy/bitbucket/rest/features/InsightsApiMockTest.java

```diff
@@ -0,0 +1,478 @@
+
+    private final String getMethod = "GET";
+    private final String deleteMethod = "DELETE";
+    private final String putMethod = "PUT";
+
+    private final String restApiPath = "/rest/insights/";
+
+    private final String projectsKeyword = "/projects/";
+    private final String repoKeyword = "/repos/";
+    private final String commitKeyword = "/commits/";
+    private final String annotationsKeyword = "/annotations";
+    private final String reportsKeyword = "/reports";
+
+
+    private final String qwertyKeyword = "qwerty";
+    private final String limitKeyword = "limit";
+    private final String startKeyword = "start";
+    private final String annotationsJsonFile = "/annotations.json";
+    private final String mockPath = ".gitignore";
+
+    public void testListReports() throws Exception {
+        final MockWebServer server = mockWebServer();
+
+        server.enqueue(new MockResponse().setBody(payloadFromResource("/insight-report-page.json")).setResponseCode(200));
+        final BitbucketApi baseApi = api(server.getUrl("/"));
+        final InsightsApi api = baseApi.insightsApi();
+        try {
+
+            final InsightReportPage insightReportPage = api.listReports(projectKey, repoKey, commitHash, 100, 0);
+            assertThat(insightReportPage).isNotNull();
+            assertThat(insightReportPage.values()).isNotEmpty();
+            assertThat(insightReportPage.errors()).isEmpty();
+
+            final Map<String, ?> queryParams = ImmutableMap.of(limitKeyword, 100, startKeyword, 0);
+
+            assertSent(server,
+                       getMethod,
+                       restApiPath + BitbucketApiMetadata.API_VERSION
+                           + projectsKeyword + projectKey + repoKeyword + repoKey + commitKeyword + commitHash + reportsKeyword, queryParams);
+        } finally {
+            baseApi.close();
+            server.shutdown();
+        }
+    }
+
+    public void testListReportsOnError() throws Exception {
+        final MockWebServer server = mockWebServer();
+
+        server.enqueue(new MockResponse().setBody(payloadFromResource("/repository-not-exist.json")).setResponseCode(404));
+        final BitbucketApi baseApi = api(server.getUrl("/"));
+        final InsightsApi api = baseApi.insightsApi();
+        try {
+
+            final InsightReportPage insightReportPage = api.listReports(projectKey, repoKey, commitHash, 100, 0);
+            assertThat(insightReportPage).isNotNull();
+            assertThat(insightReportPage.values()).isEmpty();
+            assertThat(insightReportPage.errors()).isNotEmpty();
+
+            final Map<String, ?> queryParams = ImmutableMap.of(limitKeyword, 100, startKeyword, 0);
+            assertSent(server,
+                       getMethod,
+                       restApiPath + BitbucketApiMetadata.API_VERSION
+                           + projectsKeyword + projectKey + repoKeyword + repoKey + commitKeyword + commitHash + reportsKeyword, queryParams);
```

Comment:
:warning: Line is longer than 140 characters (found 142). (#QR127:108#)-CHECKSTYLE_LineLengthCheck